lib/bolt/project.rb

```diff
@@ -45,20 +46,34 @@ def initialize(path, type = 'option')
       @resource_types = @path + '.resource_types'
       @type = type
 
-      project_file = @path + 'project.yaml'
-      @data = Bolt::Util.read_optional_yaml_hash(File.expand_path(project_file), 'project') || {}
+      @project_file = @path + 'project.yaml'
+      @data = Bolt::Util.read_optional_yaml_hash(File.expand_path(@project_file), 'project') || {}
       validate
     end
 
     def to_s
       @path.to_s
     end
 
+    # This API is used to prepend the project as a module to Puppet's internal
+    # module_references list. CHANGE AT YOUR OWN RISK
+    def to_h
+      { path: @path, name: name }
+    end
+
     def eql?(other)
       path == other.path
     end
     alias == eql?
 
+    def file?
+      File.exist?(@project_file)
+    end
+
+    def name
+      @data['name']
```

Comment:
```suggestion
      @data['name'] || File.basename(@path)
```

Puppet is expecting a project to have a name: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/node/environment.rb#L302-L303

If `name` isn't set in `project.yaml`, then it should pick up the name of the directory itself. Otherwise Puppet sees the project having a name of `nil`, which isn't a valid module:

```
        Invalid module name ''; module names must match either:
        An installed module name (ex. modulename) matching the expression /^[a-z][a-z0-9_]*$/ -or-
        A namespaced module name (ex. author-modulename) matching the expression /^[a-zA-Z0-9]+[-][a-z][a-z0-9_]*$/
```